.github/.copilot-instructions.md

```diff
@@ -0,0 +1,112 @@
+# Copilot Instructions for meshStack Terraform Provider
+
+## Repository Overview
+
+This is the official **meshStack Terraform Provider** developed by meshcloud GmbH. It enables Infrastructure as Code (IaC) management of meshStack resources through Terraform by integrating with the meshStack meshObject API.
+
+**Key Information:**
+- **Purpose**: Manage meshStack cloud resources (projects, workspaces, tenants, building blocks, etc.) using Terraform
+- **API Integration**: Uses the meshStack meshObject API (`/api/meshobjects` endpoints)
+- **API Documentation**: https://docs.dev.meshcloud.io/api/index.html#mesh_objects
+- **Provider Registry**: https://registry.terraform.io/providers/meshcloud/meshstack/latest/docs
+- **License**: MPL-2.0
+
+## Architecture Overview
+
+This provider follows the standard Terraform Provider Plugin Framework pattern:
+
+```
+├── main.go                 # Provider entry point
+├── internal/provider/      # Provider implementation (resources, data sources)
+├── client/                 # API client for meshStack meshObject API
+├── docs/                   # Terraform registry documentation
+├── examples/               # Example Terraform configurations
+└── templates/              # Documentation templates
+```
+
+### API Authentication
```

Comment:
d: not sure if this is wasted context (auth is already codified in the provider)

Reply:
a more useful explanation here would be the key concept of workspace and that most resources live in a workspace, and that API tokens are also usually scoped to a single workspace